docs/npm_nodejs.md

```diff
@@ -0,0 +1,59 @@
+# Using npm/Node.js code
+
+While Deno is pretty powerful itself, there is a large eco-system of code in the
+[npm](https://npmjs.com/) registry, and many people will want to re-leverage
+tools, code and libraries that are built for [Node.js](https://nodejs.org/). In
+this chapter we will explore how to use it.
+
+The good news, is that in many cases, it _just works_.
+
+There are some foundational things to understand about differences between
+Node.js and Deno that can help in understanding that challenges might be faced:
+
+- Current Node.js supports both CommonJS and ES Modules, while Deno only
+  supports ES Modules. The addition of stabilized ES Modules in Node.js is
+  relatively recent and most code written for Node.js is in the CommonJS format.
+- Node.js has quite a few built-in modules that can be imported and they are a
+  fairly expansive set of APIs. Deno has no built-in modules and only provides a
+  single global namespace/variable named `Deno` and it is part of our goals to
+  try to keep the APIs there to a minimum. Lots of code written for Node.js
+  expects/depends upon these built-in APIs to be available.
+- Node.js has a "magical" module resolution algorithm, where you can import
+  bare-specifiers (e.g. `react` or `lodash`) and Node.js will look in your local
+  and global `node_modules` for a path, introspect the `package.json` and try to
+  see if there is a module named the right way. Deno has no "magical"
+  resolution, much in the way a web browser doesn't have "magical" resolution.
+  For local files, Deno expects a full module name, including the extension.
+  When dealing with remote imports, Deno expects the web server to do any
+  "resolving" and provide back the media type of the code (just like a
+  web-browser).
```

Comment:
Worth noting that we fallback to checking file extension if specific content type is missing?